mayastor/src/rebuild.rs

```diff
@@ -195,24 +203,25 @@ impl RebuildTask {
 
     /// Rebuilds a non-healthy child from a healthy child from start to end
     async fn run(&mut self) {
-        self.state = RebuildState::Running;
+        self.change_state(RebuildState::Running);
         self.current = self.start;
         self.stats();
 
         while self.current < self.end {
             if let Err(e) = self.copy_one().await {
                 error!("Failed to copy segment {}", e);
-                self.state = RebuildState::Failed;
+                self.change_state(RebuildState::Failed);
                 self.send_complete();
```

Comment:
not changed by your PR but just noticed, are we missing a return here?

Reply:
Good spot - done